gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -67,6 +67,10 @@ def initialize(method:, method_name:, raw_arg_types:, raw_return_type:, bind:, m
     if !extra_names.empty?
       raise "The declaration for `#{method.name}` has extra parameter(s): #{extra_names.join(', ')}"
     end
+    
```

Comment:
TIL that `Array#-` does set difference (on the line above `param_names - declared_param_names`)